Kajabi/sage-lib
# Transaction card updates

## Description
The transaction card design has been updated. This PR updates the component to match these changes.

- Removes dropdown
- Adds option to add a link to the `name` through `name_href` attribute
- Renames `product` element and attributes to `related_property`
- Adds option to add a link to the `related_property` through `related_property_href` attribute
- Adds `transaction_state` section (replaces dropdown)
- Adds `name_tag` (rails) `NameTag` (react) to allow for heading element to be set and not be hard-coded. Will default to h4.


### Screenshots

|  Before  |  After  |
|--------|--------|
|![Screen Shot 2021-04-26 at 12 42 00 PM](https://user-images.githubusercontent.com/1175111/116140906-d985f800-a68c-11eb-813a-55e16df35675.png)|![Screen Shot 2021-04-26 at 12 41 36 PM](https://user-images.githubusercontent.com/1175111/116140922-e0ad0600-a68c-11eb-8631-653c6000ac8b.png)|


## Test notes

REACT:
- Visit: http://localhost:4100/?path=/story/sage-transactioncard--default
 - Verify dropdown is no longer available
 - Verify link is added to the `name`
 - Verify link is added to the `related_property`
 - Verify `transaction_state` text displays in place of dropdown
 - Verify product-related naming is updated to `related_property` (elements and attributes)
 - Adjust controls to ensure things work as expected.

RAILS: 
 - Visit http://localhost:4000/pages/object/transaction_card
 - Verify dropdown is no longer available
 - Verify link is added to the `name`
 - Verify link is added to the `related_property`
 - Verify `transaction_state` text displays in place of dropdown
 - Verify product-related naming is updated to `related_property` (elements and attributes)

### Estimated impact
- [ ] (**LOW**) Removes dropdown, adds an optional link to name and related_property, and adds new transaction_state text for Transaction Card. This component is not currently in use and these changes are being made in preparation for work to begin during the next cycle.


## Related
